routers/web/repo/issue.go

```diff
@@ -1728,6 +1753,28 @@ func UpdateIssueTitle(ctx *context.Context) {
 	})
 }
 
+// IssuePrivate sets the issue confidential
+func IssuePrivate(ctx *context.Context) {
+	issue := GetActionIssue(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if !ctx.IsSigned || (!issue.IsPoster(ctx.User.ID) && !ctx.Repo.CanWriteIssuesOrPulls(issue.IsPull)) {
```

Comment:
Is `CanWriteIssuesOrPulls` the correct permission?
I don't think so.
I think, here, we should be really restrictive and only allow repo admins to change the state of these issues.

Reply:
> Is `CanWriteIssuesOrPulls` the correct permission?
> I don't think so.

I haven't really started about the restrictive permission - The `CanWriteIssuesOrPulls` is currently as a "placebo" until a better restrictive(like as you suggest the repo admins)

Reply:
I do think that the issue's poster(maybe not even?) + Repo admins should be the only ones that is able to change the `private` state.

Reply:
Changed.